Encryption.md

```diff
@@ -133,48 +154,144 @@ Algorithms section. The length of the encryption buffer (a 4-byte little endian)
 written to the output stream, followed by the buffer itself.
 
 The column pages (data and dictionary) are encrypted after compression. For each page, 
-the encryption buffer is comprised of an IV, ciphertext and (in case of AES_GCM_V1) of a 
+the encryption buffer is comprised of an IV (nonce), ciphertext and (in case of AES_GCM_V1) of a 
 tag, as described in the Algorithms section. Only the buffer is written to the output 
-stream - not need to write the length of the buffer, since the length (size of the page after
+stream - no need to write the length of the buffer, since the length (page size after
 compression and encryption) is kept in the page headers.
 
-A `crypto_meta_data` field in set in each `ColumnChunk` in the encrypted columns. 
+A `crypto_meta_data` field is set in each `ColumnChunk` in the encrypted columns. 
 `ColumnCryptoMetaData` is a union - the actual structure is chosen depending on whether the 
 column is encrypted with the footer key, or with a column-specific key. For the latter, 
 a key metadata can be specified, with a maximal length of 512. Key metadata is a free-form
-byte array that can be used by a reader to retrieve the column encryption key. 
-
-Parquet file footer, and its nested structures, contain sensitive information - ranging 
+byte array that can be used by a reader to retrieve the encryption key (see next subsection). 
+
+```c
+struct EncryptionWithFooterKey {
+}
+
+struct EncryptionWithColumnKey {
+  /** Column path in schema **/
+  1: required list<string> path_in_schema
+  
+  /** Retrieval metadata of the column-specific key **/
+  2: optional binary column_key_metadata
+}
+
+union ColumnCryptoMetaData {
+  1: EncryptionWithFooterKey ENCRYPTION_WITH_FOOTER_KEY
+  2: EncryptionWithColumnKey ENCRYPTION_WITH_COLUMN_KEY
+}
+
+struct ColumnChunk {
+...
+  /** Crypto metadata of encrypted columns **/
+  8: optional ColumnCryptoMetaData crypto_meta_data
```

Comment:
(nit) Just noticed that this field is named *_meta_data, while other structs use *_metadata. You may consider renaming this field here and in the proto itself.

Reply:
just noticed that the current FileMetaData structure has a field named key_value_metadata. So there are already metadata and meta_data fields..

Reply:
This change still needs to be made.